mlflow/pyfunc/__init__.py

```diff
@@ -946,12 +959,18 @@ def load_model(
 
     _add_code_from_conf_to_system_path(local_path, conf, code_key=CODE)
     data_path = os.path.join(local_path, conf[DATA]) if (DATA in conf) else local_path
+
+    if isinstance(model_config, str):
+        model_config = _validate_and_get_model_config_from_file(model_config)
+
+    # conf.get(MODEL_CONFIG) this should ALWAYS be a dict (no more saving as file!)
     model_config = _get_overridden_pyfunc_model_config(
         conf.get(MODEL_CONFIG, None), model_config, _logger
     )
-
     try:
-        if model_config:
+        # currently, langchain model_config must be specified using ModelConfig() class in code
+        # TODO: support loading langchain with model_config using _load_pyfunc
+        if model_config and "langchain" not in conf[MAIN]:
```

Comment:
Instead of adding a special case for a flavor in `mlflow.pyfunc.load_model()`, can we add the `model_config` parameter to `mlflow.langchain._load_pyfunc` as a no-op and add the TODO there to use the supplied `model_config`?

Special cases for flavors in `mlflow.pyfunc.load_model` is an antipattern